### PR TITLE
Implementing Lastpathcomponent

### PR DESCRIPTION
**Pull request by aballway**
Following up on Jeyaram's PR #870 this adds some more test cases and fixes a bug in _CFStartOfLastPathComponent and _CFStartOfLastPathComponent2 which were not properly handling multiple trailing slashes ("/foo/bar//")

### CODE REVIEW

**Comment by MSFTFox** [comment]
:shipit:

**Comment by rajsesh-msft** [comment]
:shipit:

File: Frameworks/CoreFoundation/Base.subproj/CFFileUtilities.c [hunk 1]
```diff
@@ -988,7 +988,8 @@ CF_PRIVATE CFIndex _CFStartOfLastPathComponent(UniChar *unichars, CFIndex length
         return 0;
     }
     for (idx = length - 1; idx; idx--) {
-        if (IS_SLASH(unichars[idx - 1])) {
+        // WinObjC change - handle cases of multiple trailing slashes
```
Comment (rajsesh-msft) [inline Frameworks/CoreFoundation/Base.subproj/CFFileUtilities.c]: >WinObjC [](start = 11, length = 7)

nit: WINOBJC in all caps.

File: Frameworks/CoreFoundation/Base.subproj/CFFileUtilities.c [hunk 2]
```diff
@@ -988,7 +988,8 @@ CF_PRIVATE CFIndex _CFStartOfLastPathComponent(UniChar *unichars, CFIndex length
         return 0;
     }
     for (idx = length - 1; idx; idx--) {
-        if (IS_SLASH(unichars[idx - 1])) {
+        // WINOBJCchange - handle cases of multiple trailing slashes
```
Comment (DHowett-MSFT) [inline Frameworks/CoreFoundation/Base.subproj/CFFileUtilities.c]: >WINOBJCchange [](start = 11, length = 13)

nit: `WINOBJC`

**Comment by DHowett-MSFT** [comment]
:shipit:

File: tests/unittests/Foundation/NSStringTests.m [hunk 3]
```diff
@@ -607,4 +607,72 @@ void testUrlCharacterSetEncoding(NSString* decodedString, NSString* encodedStrin
...
+    EXPECT_OBJCEQ(@".", string.lastPathComponent);
+
+    string = @"/baz/bar/.foo";
+    EXPECT_OBJCEQ(@".foo", string.lastPathComponent);
+
+    string = @"/tmp/foo.";
+    EXPECT_OBJCEQ(@"foo.", string.lastPathComponent);
+
+    string = @"/tmp/foo///";
+    EXPECT_OBJCEQ(@"foo", string.lastPathComponent);
```
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSStringTests.m]: >EXPECT_OBJCEQ [](start = 4, length = 13)

Really? That scares me a little

File: Frameworks/CoreFoundation/Base.subproj/CFFileUtilities.c (hunk 2 above)
Comment (aballway) [inline Frameworks/CoreFoundation/Base.subproj/CFFileUtilities.c]: looks like I undid this partway, oops

---
In reply to: [76706890](https://github.com/Microsoft/WinObjC/pull/934#discussion_r76706890) [](ancestors = 76706890)

File: tests/unittests/Foundation/NSStringTests.m (hunk 3 above)
Comment (aballway) [inline tests/unittests/Foundation/NSStringTests.m]: Double checked on reference platform.  Not falling for that again

---
In reply to: [76707059](https://github.com/Microsoft/WinObjC/pull/934#discussion_r76707059) [](ancestors = 76707059)

**Pull request merged**